pennylane/ops/pulse/convenience_functions.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This file contains the ``piecewise`` pulse convenience function."""
+has_jax = True
+try:
+    import jax.numpy as jnp
+except ImportError:
+    has_jax = False
+
+
+def rect(x, ti, tf):
```

Comment:
Can you generalize this such that it inputs `cond (Union[tuple, tuple(tuple)]` I. E. You can provide more than one time window where the function is active?

Reply:
Done!